### DISCUSSION TITLE
Zmiany, o które prosiliście, w wersji v3.39

### DISCUSSION THREAD

**Discussion by TomaszGasior** [TomaszGasior/RadioLista-v3#173] [0. Ogłoszenia Announcements]
Jutro wieczorem wdrożę na radiolista.pl wersję v3.39 serwisu. **Wystąpi przerwa techniczna do ok. pół godziny.** Pojawi się kilka zmian, które, mam nadzieję, spełnią wasze oczekiwania. :)

Podpowiedzi nazw stacji i lokalizacji nadajników przy edycji stacji
---

Formularz edytowania/dodawania stacji radiowej podpowiada możliwe wartości przy uzupełnianiu pól takich jak nazwa czy lokalizacja. W nowej wersji RadioListy (zadanie #97) dotychczasowe podpowiedzi wpisane „na sztywno” (https://github.com/TomaszGasior/RadioLista-v3/commit/a8a6fd867bd81b34d5a3614503da4c08fe9aefc4) i wymyślone odgórnie przeze mnie zostały zastąpione dynamicznym mechanizmem, który podpowiedzi do danego pola generuje na podstawie waszych własnych preferencji, tj. na podstawie dotychczas wpisanych wartości do danego pola w innych stacjach tego samego wykazu.

Nowe podpowiedzi działają dla następujących pól:
- nazwa,
- lokalizacja nadajnika
- grupa medialna,
- multipleks cyfrowy,
- kraj,
- region.

Ze względu na optymalizację obciążenia systemu podpowiedzi będą sugerowane tylko dla tych kolumn, które w ustawieniach wykazu są włączone.

Większa swoboda ustawień wyglądu wiersza wykazu
---

Zgodnie  z prośbą @UniqueTM2 w wątku #163 dodałem większą swobodę w ustawieniach wyglądu stacji radiowych. Od teraz kolor tła oraz pogrubienie, pochylenie, a także przekreślenie można ustawiać oddzielnie od siebie.

W zadaniu #164 planowałem pierwotnie dodać także dodatkowe kolory obok dotychczas dostępnych. Zrezygnowałem z tego pomysłu dla oszczędności czasu, ale dzięki wdrożonym zmianom dodanie kilku nowych kolorów tła wierszy będzie dla mnie prostsze. Jeśli macie taką potrzebę, proszę o propozycję konkretnych kolorów. Maksymalnie kilka z nich możemy wdrożyć. (Żeby wygenerować kolor można użyć: https://www.google.com/search?q=hex+color+picker ).

Uwaga: ze względu na błąd #170 czcionka może wyglądać niespójnie przy jednoczesnym pogrubieniu i pochyleniu.

Uproszczenie kolumny „lokalność”
---

Skomplikowana w obsłudze i często używana niepoprawnie kolumna „lokalność” zostanie zastąpiona nową i prostszą kolumną „region”, która będzie zwykłym potem tekstowym podobnie jak „kraj”. Można tam będzie wpisać np. nazwę województwa lub miasta, dla którego nadaje stacja np. pasma lokalne. Sens „regionu” jest podobny jak „lokalności”, ale nowa kolumna jest prostsza w obsłudze i w utrzymaniu oraz bardziej uniwersalna.

Szczegóły zmiany: #168 #160 

Pozostałe zmiany
---

- Kolumna RDS PI nie wyszarza się już w formularzu edycji/dodawania stacji, gdy kolumna RDS jest wyłączona w wykazie. #169 
- Po rocznej przerwie od teraz znowu w ustawieniach wyglądu wykazu można zmieniać kolor tła i kolor tekstu w sposób graficzny, jednym kliknięciem. #126

**Comment by TomaszGasior** [comment]
Zmiany zostały przed chwilą wdrożone na radiolista.pl.

**Comment by HyperDX** [comment]
Nowe funkcje jak i zmiany bardzo cieszą! Zwłaszcza dziękuję za wdrożenie nowych opcji formatowania wyglądu wiersza. Co do nowych kolorów to mam parę propozycji: #FFFF00 oraz #FF00FF. Mimo, iż nie mam potrzeby stosowania większej ilości różnych kolorów zaznaczeń to myślałem też nad zastąpieniem aktualnych kolorów nowymi, które mogą być wprowadzone i pewnie inni użytkownicy też by dodali chętnie jakieś propozycje od siebie. ;)

**Comment by UniqueTM2** [comment]
Popieram. Te kolory sie przydadza

**Comment by TomaszGasior** [comment]
Dwa nowe kolory zostały dodane w ramach zadania #174  w wersji v3.40, którą przed chwilą wdrożyłem na radiolista.pl. Naprawiłem też błąd #170 związany z czcionką, wspomniany na początku tego wątku.
